pytorch_lightning/utilities/batch_loop_common.py

```diff
@@ -0,0 +1,105 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
pytorch_lightning/utilities doesn't feel like right place for this; it's more user facing, while these functions here are intended to be used inside of the trainer's loops

shall we add a trainer/utilities directory for these sorts of helpers? @tchaton @awaelchli @carmocca?

Reply:
Why not a loop/utilities?